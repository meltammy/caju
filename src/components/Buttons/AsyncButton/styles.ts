import styled from "styled-components";
import { Spinner } from "~/components/Spinner";

export const StyledSpinner = styled(Spinner)`
  width: 5px;
  height: 5px;
  border-width: 3px;
`;
