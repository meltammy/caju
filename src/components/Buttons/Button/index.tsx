import styled from "styled-components";

const Button = styled.button<{ $variant: "outline" | "filled"; color: string }>`
  outline: none;
  display: flex;
  align-items: center;
  border: none;
  border-radius: 1rem;
  padding: 0.5rem 2rem;
  place-content: center;
  cursor: pointer;
  height: 3.5rem;
  box-shadow: rgba(149, 157, 165, 0.2) 0px 0.5rem 1.5rem;
  font-size: 1rem;
  font-weight: 600;
  text-wrap: nowrap;

  &:disabled {
    background-color: ${({ theme }) => theme.colors.grey};
    opacity: 1;
  }

  ${(props) => {
    switch (props.$variant) {
      case "outline":
        return `
          border: 2px solid ${props.color};
          color:  ${props.color};
          background: transparent;
        `;
      case "filled":
        return `
          background-color:  ${props.color};
          color: ${props.theme.colors.white};
        `;
      default:
        return `
          background-color: ${props.color ?? props.theme.colors.green};
          color: ${props.theme.colors.white};
        `;
    }
  }}
`;

export default Button;
