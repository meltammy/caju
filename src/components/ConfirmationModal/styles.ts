import styled from "styled-components";

export const ModalOverlay = styled.div<{ $isOpen: boolean }>`
  display: ${(props) => (props.$isOpen ? "flex" : "none")};
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: rgba(0, 0, 0, 0.5);
  justify-content: center;
  align-items: center;
`;

export const ModalContainer = styled.div`
  background: white;
  border-radius: 8px;
  padding: 20px;
  min-width: 500px;
  max-width: 95vw;
  box-shadow: 0 2px 10px rgba(0, 0, 0, 0.1);
  z-index: 999;

  @media screen and (max-width: 1200px) {
    min-width: 50vw;
    margin: 1rem;
  }
`;

export const Title = styled.h2`
  margin: 0;
`;

export const Description = styled.p`
  margin: 10px 0 20px;
`;

export const ButtonContainer = styled.div`
  display: flex;
  gap: 1rem;

  @media screen and (max-width: 1200px) {
    flex-direction: column-reverse;
  }

  button {
    width: 100%;
    place-content: center;
  }
`;
