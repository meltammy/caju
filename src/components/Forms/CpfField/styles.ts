import ReactInputMask from "react-text-mask";
import styled from "styled-components";

export const Input = styled(ReactInputMask)`
  width: 100%;
  padding: 0 0.5rem;
  min-height: 2.25rem;
  background-color: ${({ theme }) => theme.colors.white};
  transition: all 0.2s ease-in-out 0s;

  vertical-align: middle;
  border: 1px solid rgba(36, 28, 21, 0.3);
  border-radius: 0.5rem;

  font-size: 1rem;
  line-height: 1.125rem;
  font-weight: normal;

  &:focus {
    border: 1px solid ${({ theme }) => theme.colors.petroleumBlue};
    box-shadow: inset 0 0 0 1px ${({ theme }) => theme.colors.petroleumBlue};
  }
`;
