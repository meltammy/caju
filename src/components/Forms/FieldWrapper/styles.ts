import styled from "styled-components";

export const ErrorMessage = styled.span`
  color: ${({ theme }) => theme.colors.primary};
  font-size: 0.75rem;
`;

export const Container = styled.div`
  width: 100%;
`;

export const Label = styled.label`
  width: 100%;
  font-weight: 600;
`;
