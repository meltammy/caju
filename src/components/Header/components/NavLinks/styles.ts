import styled from "styled-components";

export const NavItems = styled.a`
  color: ${({ theme }) => theme.colors.primary};
  font-size: 1.5rem;
  font-weight: 600;
`;
