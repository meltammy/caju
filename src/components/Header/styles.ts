import styled from "styled-components";

export const Container = styled.header`
  background: ${({ theme }) => theme.colors.white};
  width: 100%;
  box-shadow: 4px 4px 16px 3px #0000001f;
  height: 4rem;
  position: fixed;
  top: 0;
  display: flex;
  align-items: center;
  padding: 0px 1.5rem;
  gap: 2rem;
  z-index: 1;
`;

export const Title = styled.h1`
  color: ${({ theme }) => theme.colors.primary};
  font-size: 1.5rem;
  font-weight: 600;
`;
