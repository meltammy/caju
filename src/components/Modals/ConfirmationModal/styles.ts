import styled from "styled-components";

export const ModalContainer = styled.div`
  min-width: 32rem;
  max-width: 95vw;
  z-index: 999;
  padding: 1.5rem;
  background: white;
  border-radius: 0.5rem;
  box-shadow: 0 2px 10px rgba(0, 0, 0, 0.1);

  @media screen and (max-width: ${({ theme }) => theme.breakpoints.desktop}) {
    min-width: 50vw;
    margin: 1rem;
  }
`;

export const Title = styled.h2`
  margin: 0;
`;

export const Description = styled.p`
  margin: 10px 0 1.5rem;
`;

export const ButtonContainer = styled.div`
  display: flex;
  gap: 1rem;

  @media screen and (max-width: ${({ theme }) => theme.breakpoints.desktop}) {
    flex-direction: column-reverse;
  }
`;
