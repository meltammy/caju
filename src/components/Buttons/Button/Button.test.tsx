import "jest-styled-components";
import "@testing-library/jest-dom";

import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import Button from ".";
import { theme } from "~/../styles/theme";
import { ComponentProps } from "react";

const renderButton = (props?: ComponentProps<typeof Button>) => {
  return render(
    <ThemeProvider theme={theme}>
      <Button {...props}>{props?.children || "Click Me"}</Button>
    </ThemeProvider>
  );
};

describe("Button component", () => {
  it("should render correctly with default styles", () => {
    const { getByRole } = renderButton();

    const button = getByRole("button", { name: /click me/i });
    expect(button).toHaveStyleRule("background-color", theme.colors.green);
  });

  it("should render disabled button with grey background", () => {
    const { getByRole } = renderButton({
      children: "disabled",
      disabled: true,
    });

    const button = getByRole("button", { name: /disabled/i });

    expect(button).toBeInTheDocument();
    expect(button).toBeDisabled();
    expect(button).toHaveStyleRule("background-color", theme.colors.grey, {
      modifier: ":disabled",
    });
  });

  it("should match snapshot", () => {
    const { asFragment } = renderButton();
    expect(asFragment()).toMatchSnapshot();
  });
});
