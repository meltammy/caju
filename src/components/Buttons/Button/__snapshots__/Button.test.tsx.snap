// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  outline: none;
  display: flex;
  align-items: center;
  border: none;
  border-radius: 36px;
  padding: 8px 32px;
  background-color: #64a98c;
  cursor: pointer;
  height: 56px;
  color: #fff;
  box-shadow: rgba(149, 157, 165, 0.2) 0px 8px 24px;
  font-size: 16px;
  font-weight: 600;
}

.c0:disabled {
  background-color: #ccc;
  opacity: 1;
}

<button
    class="c0"
  >
    Click Me
  </button>
</DocumentFragment>
`;

exports[`Button component should match snapshot 1`] = `
<DocumentFragment>
  .c0 {
  outline: none;
  display: flex;
  align-items: center;
  border: none;
  border-radius: 36px;
  padding: 8px 32px;
  background-color: #64a98c;
  cursor: pointer;
  height: 56px;
  color: #fff;
  box-shadow: rgba(149, 157, 165, 0.2) 0px 8px 24px;
  font-size: 16px;
  font-weight: 600;
}

.c0:disabled {
  background-color: #ccc;
  opacity: 1;
}

<button
    class="c0"
  >
    Click Me
  </button>
</DocumentFragment>
`;
